main.go

```diff
@@ -5,58 +5,145 @@ import (
-			Argument:  "example",
-			Shorthand: "e",
+	opts = []*sensu.PluginConfigOption{
+		&sensu.PluginConfigOption{
+			Path:      host,
+			Env:       "WAVEFRONT_HOST",
+			Argument:  host,
+			Shorthand: "",
+			Default:   "127.0.0.1",
+			Usage:     "the host of the wavefront proxy",
+			Value:     &handlerConfig.Host,
+		},
+		&sensu.PluginConfigOption{
+			Path:      port,
+			Env:       "WAVEFRONT_METRICS_PORT",
+			Argument:  port,
+			Shorthand: "m",
+			Default:   2878,
+			Usage:     "the port of the wavefront proxy",
+			Value:     &handlerConfig.MetricsPort,
+		},
+		&sensu.PluginConfigOption{
+			Path:      flush,
+			Env:       "WAVEFRONT_FLUSH_INTERVAL_SECONDS",
+			Argument:  flush,
+			Shorthand: "f",
+			Default:   1,
+			Usage:     "the flush interval of the wavefront proxy (in seconds)",
+			Value:     &handlerConfig.FlushIntervalSeconds,
+		},
+		&sensu.PluginConfigOption{
+			Path:      prefix,
+			Env:       "WAVEFRONT_PREFIX",
+			Argument:  prefix,
+			Shorthand: "p",
 			Default:   "",
-			Usage:     "An example configuration option",
-			Value:     &handlerConfig.Example,
+			Usage:     "the prefix to append to the metric name",
+			Value:     &handlerConfig.Prefix,
+		},
+		&sensu.PluginConfigOption{
+			Path:      tags,
+			Env:       "WAVEFRONT_TAGS",
+			Argument:  tags,
+			Shorthand: "t",
+			Default:   nil,
+			Usage:     "the additional tags to merge with the metric tags",
+			Value:     &handlerConfig.Tags,
 		},
-	}
-
-	options = []*sensu.PluginConfigOption{
-		&handlerConfigOptions.Example,
 	}
 )
 
 func main() {
-	handler := sensu.NewGoHandler(&handlerConfig.PluginConfig, options, checkArgs, executeHandler)
+	handler := sensu.NewGoHandler(&handlerConfig.PluginConfig, opts, checkArgs, executeHandler)
 	handler.Execute()
 }
 
-func checkArgs(_ *types.Event) error {
-	if len(handlerConfig.Example) == 0 {
-		return fmt.Errorf("--example or HANDLER_EXAMPLE environment variable is required")
+func checkArgs(event *corev2.Event) error {
+	if !event.HasMetrics() {
+		return fmt.Errorf("event does not contain metrics")
 	}
 	return nil
 }
 
-func executeHandler(event *types.Event) error {
-	log.Println("executing handler with --example", handlerConfig.Example)
+func executeHandler(event *corev2.Event) error {
+	if len(event.Metrics.Points) == 0 {
+		log.Println("event does not contain metric points")
+		return nil
+	}
+
+	proxyCfg := &wavefront.ProxyConfiguration{
+		Host:                 handlerConfig.Host,
+		MetricsPort:          handlerConfig.MetricsPort,
+		FlushIntervalSeconds: handlerConfig.FlushIntervalSeconds,
+	}
+
+	sender, err := wavefront.NewProxySender(proxyCfg)
+	if err != nil {
+		return err
+	}
+
+	for _, point := range event.Metrics.Points {
+		tags := make(map[string]string)
+		for _, tag := range point.Tags {
+			tags[tag.Name] = tag.Value
+		}
+
+		// merge tags if provided as config option
+		if handlerConfig.Tags != nil {
+			for k, v := range handlerConfig.Tags {
+				tags[k] = v
+			}
+		}
```

Comment:
So unless I'm mistaken, you're giving precedence to the tags the handler has over the tags the event's metrics have. We did the opposite in the OpenTSDB handler! I'm not sure which way is "correct" now!

If I recall correctly, we based that on what the Classic Ruby extensions were doing; I need to check again.

Reply:
Well the true metric comes from the Sensu event. The metric tag enrichment is somewhat of an override/additional configuration that happens after the event has been processed. For that reason, I chose to give precedence to the handlers tags since operators should be aware of any additional tags they're setting. 

If that's what the classic extension is doing I'm happy to change it. IMO it makes more sense to give users the ability to overwrite tags if they so choose!

Reply:
Matching the classic enterprise logic is ideal for users doing a migration.